src/components/ha-selector/ha-selector-target.ts

```diff
@@ -58,8 +61,9 @@ export class HaTargetSelector extends SubscribeMixin(LitElement) {
     if (changedProperties.has("selector")) {
       const oldSelector = changedProperties.get("selector");
       if (
-        oldSelector !== this.selector &&
-        this.selector.target.device?.integration
+        (oldSelector !== this.selector &&
+          this.selector.target.device?.integration) ||
+        this.selector.target.entity?.integration
```

Comment:
```suggestion
        (oldSelector !== this.selector &&
          this.selector.target.device?.integration) ||
        (oldSelector !== this.selector && this.selector.target.entity?.integration)
```
Shouldn't this be checked on both sides?